editor/src/components/canvas/controls/svg-control-component.tsx

```diff
@@ -164,22 +163,24 @@ const controlToSvg = (control: SvgFragmentControl, index: number) => {
       ),
       RU.create(
         control.type,
-        R.merge(control.props, {
+        {
+          ...control.props,
           id: `control-${control.controlid}`,
           fill: `url(#${patternId})`,
           key: `control-${control.controlid}-${index}`,
-        }),
+        } as any,
```

Comment:
have you tried `as const` ? sometimes it makes React realize that we are using the proper string literals…

Reply:
This appeared to be because of the `id` field, I just attempted to maintain what was already there.